test/extensions/stats_sinks/metrics_service/grpc_metrics_service_impl_test.cc

```diff
@@ -282,6 +282,44 @@ TEST_F(MetricsServiceSinkTest, ReportMetricsWithTags) {
   sink.flush(snapshot_);
 }
 
+TEST_F(MetricsServiceSinkTest, DefaultFlushPredicate) {
+  MetricsFlusher flusher(true, true);
+
+  auto used_counter = std::make_shared<NiceMock<Stats::MockCounter>>();
```

Comment:
nit: suggest a helper function to factor out the 5-line stanzas making the mock counters (used 4x so far).

Reply:
Added out helper functions to all the metrics types since I was in here, I think this helped the overall readability of the whole file